activation/post_supervisor.go

```diff
@@ -232,67 +225,69 @@ func (ps *PostSupervisor) runCmd(
 	postOpts PostSetupOpts,
 	provingOpts PostProvingOpts,
 ) error {
-	for {
-		args := []string{
-			"--address", cmdCfg.NodeAddress,
+	args := []string{
+		"--address", cmdCfg.NodeAddress,
 
-			"--k1", strconv.FormatUint(uint64(postCfg.K1), 10),
-			"--k2", strconv.FormatUint(uint64(postCfg.K2), 10),
-			"--k3", strconv.FormatUint(uint64(postCfg.K3), 10),
-			"--pow-difficulty", postCfg.PowDifficulty.String(),
+		"--k1", strconv.FormatUint(uint64(postCfg.K1), 10),
+		"--k2", strconv.FormatUint(uint64(postCfg.K2), 10),
+		"--k3", strconv.FormatUint(uint64(postCfg.K3), 10),
+		"--pow-difficulty", postCfg.PowDifficulty.String(),
 
-			"--dir", postOpts.DataDir,
-			"-n", strconv.FormatUint(uint64(postOpts.Scrypt.N), 10),
-			"-r", strconv.FormatUint(uint64(postOpts.Scrypt.R), 10),
-			"-p", strconv.FormatUint(uint64(postOpts.Scrypt.P), 10),
+		"--dir", postOpts.DataDir,
+		"-n", strconv.FormatUint(uint64(postOpts.Scrypt.N), 10),
+		"-r", strconv.FormatUint(uint64(postOpts.Scrypt.R), 10),
+		"-p", strconv.FormatUint(uint64(postOpts.Scrypt.P), 10),
 
-			"--threads", strconv.FormatUint(uint64(provingOpts.Threads), 10),
-			"--nonces", strconv.FormatUint(uint64(provingOpts.Nonces), 10),
-			"--randomx-mode", provingOpts.RandomXMode.String(),
+		"--threads", strconv.FormatUint(uint64(provingOpts.Threads), 10),
+		"--nonces", strconv.FormatUint(uint64(provingOpts.Nonces), 10),
+		"--randomx-mode", provingOpts.RandomXMode.String(),
 
-			"--watch-pid", strconv.Itoa(os.Getpid()),
-		}
-		if cmdCfg.CACert != "" {
-			args = append(args, "--ca-cert", cmdCfg.CACert)
-		}
-		if cmdCfg.Cert != "" {
-			args = append(args, "--cert", cmdCfg.Cert)
-		}
-		if cmdCfg.Key != "" {
-			args = append(args, "--key", cmdCfg.Key)
-		}
+		// "--watch-pid", strconv.Itoa(os.Getpid()),
```

Comment:
@poszu I think there is a bug in the Post service: If I use both `--watch-pid` and `--max-retries` the service will print "shutting down" after the specified number of retries, but won't exit.

Reply:
You are right. I fixed it in https://github.com/spacemeshos/post-rs/pull/155.